src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesImplementation.cs

```diff
@@ -34,12 +69,23 @@ public sealed override void GenerateIV()
 
         public sealed override void GenerateKey()
         {
-            Key = RandomNumberGenerator.GetBytes(KeySize / BitsPerByte);
+            Span<byte> key = stackalloc byte[KeySize / BitsPerByte];
+            RandomNumberGenerator.Fill(key);
+            SetKeyCore(key);
         }
 
         protected sealed override void Dispose(bool disposing)
         {
             base.Dispose(disposing);
+            _keyBox?.Dispose();
```

Comment:
Actually, one final comment - shouldn't we only be doing the keyBox dispose if `disposing` is true? We don't want to do this disposal in a finalizer because the keyBox's own finalizer will do the resource cleanup.

Reply:
Nothing says "memento mori" than deciding you can knock out an easy change in a day.

Reply:
/me looks up "memento mori" sees it being translated as "remember that you (must eventually) die", and backpedals a bit... only intending "rememberest: thou art mortal"

Reply:
Tomato tomato.